### DISCUSSION TITLE
SAMD21 Support for Deep Sleep Mode

### DISCUSSION THREAD

**Discussion by psimonte** [zephyrproject-rtos/zephyr#50155] [Q&A]
Hello,
I am currently working on a project that uses Zephyr. The platform is the MCU SAMD21 from Microchip. Although this is well supported, driver support for deep sleep mode is unfortunately missing.

The system is battery powered, so the sleep function is essential for this application. I have tried to write the sleep driver several times, unfortunately without success. 

I want the system to go into deep sleep mode using the function k_sleep(K_MSEC(1000)), and wake up again by pressing a key before the time is up.

In my test setup I measure via a multimeter how much current is drawn from the battery. During the above mentioned sleep function about 7.5 mA flow. According to the data sheet of the SAMD21 the consumption should be in the µA range. In a bare metal setup via Microchip's IDE "Microchip Studio" I can verify this. Only under Zephyr I have no success.

Attached I send you the device tree of the system and the driver code.

The driver is located under soc/arm/atmel_sam0/samd21/power.c

with kind regards 
Simon T.

[uprogrammer.zip](https://github.com/zephyrproject-rtos/zephyr/files/9547851/uprogrammer.zip)

**Comment by nandojve** [comment]
Hi @psimonte ,

The SAMD family is supported by community. You are welcome to send contributions. Please, look at https://docs.zephyrproject.org/latest/contribute/index.html for more information about how to do it.

You are invited to reach people on Zephyr discord channel too. For more info access https://zephyrproject.org/community/#connect.
